tests/Unit/PointwiseFunctions/GeneralRelativity/Test_Ricci.cpp

```diff
@@ -7,123 +7,136 @@
+  CHECK_ITERABLE_APPROX(ricci.get(1, 1),
+                        make_with_value<DataType>(used_for_size, -178.));
+  CHECK_ITERABLE_APPROX(ricci.get(1, 2),
+                        make_with_value<DataType>(used_for_size, 143.5));
+  CHECK_ITERABLE_APPROX(ricci.get(2, 2),
+                        make_with_value<DataType>(used_for_size, -201.));
 }
 
-void test_3d_spacetime_ricci(const DataVector& used_for_size) {
+template <typename DataType>
+void test_3d_spacetime_ricci(const DataType& used_for_size) {
   constexpr size_t spatial_dim = 3;
   const auto ricci = gr::ricci_tensor(
-      make_spacetime_christoffel_second_kind<spatial_dim>(0.),
-      make_deriv_spacetime_christoffel_second_kind<spatial_dim>(0.));
-  CHECK(ricci.get(0, 0) == approx(-1928.));
-  CHECK(ricci.get(0, 1) == approx(-700.));
-  CHECK(ricci.get(0, 2) == approx(283.));
-  CHECK(ricci.get(0, 3) == approx(940.));
-  CHECK(ricci.get(1, 1) == approx(-1300.));
-  CHECK(ricci.get(1, 2) == approx(82.5));
-  CHECK(ricci.get(1, 3) == approx(968.));
-  CHECK(ricci.get(2, 2) == approx(-629.));
-  CHECK(ricci.get(2, 3) == approx(335.5));
-  CHECK(ricci.get(3, 3) == approx(-1176.));
-
-  check_tensor_doubles_equals_tensor_datavectors(
-      gr::ricci_tensor(
-          make_spacetime_christoffel_second_kind<spatial_dim>(used_for_size),
-          make_deriv_spacetime_christoffel_second_kind<spatial_dim>(
-              used_for_size)),
-      ricci);
+      make_spacetime_christoffel_second_kind<spatial_dim, DataType>(
+          used_for_size),
+      make_deriv_spacetime_christoffel_second_kind<spatial_dim, DataType>(
+          used_for_size));
+  CHECK_ITERABLE_APPROX(ricci.get(0, 0),
+                        make_with_value<DataType>(used_for_size, -1928.));
+  CHECK_ITERABLE_APPROX(ricci.get(0, 1),
+                        make_with_value<DataType>(used_for_size, -700.));
+  CHECK_ITERABLE_APPROX(ricci.get(0, 2),
+                        make_with_value<DataType>(used_for_size, 283.));
+  CHECK_ITERABLE_APPROX(ricci.get(0, 3),
+                        make_with_value<DataType>(used_for_size, 940.));
+  CHECK_ITERABLE_APPROX(ricci.get(1, 1),
+                        make_with_value<DataType>(used_for_size, -1300.));
+  CHECK_ITERABLE_APPROX(ricci.get(1, 2),
+                        make_with_value<DataType>(used_for_size, 82.5));
+  CHECK_ITERABLE_APPROX(ricci.get(1, 3),
+                        make_with_value<DataType>(used_for_size, 968.));
+  CHECK_ITERABLE_APPROX(ricci.get(2, 2),
+                        make_with_value<DataType>(used_for_size, -629.));
+  CHECK_ITERABLE_APPROX(ricci.get(2, 3),
+                        make_with_value<DataType>(used_for_size, 335.5));
+  CHECK_ITERABLE_APPROX(ricci.get(3, 3),
+                        make_with_value<DataType>(used_for_size, -1176.));
 }
 }  // namespace
 
 SPECTRE_TEST_CASE("Unit.PointwiseFunctions.GeneralRelativity.Ricci.",
                   "[PointwiseFunctions][Unit]") {
+  constexpr double d(4.0);
```

Comment:
replace 4.0 with std::numeric_limits<double>::signaling_nan()

Reply:
Why?

Reply:
I did this and it doesn't compile. What do I need to include? Tried googling but couldn't figure it out.

Reply:
Google harder ;) http://en.cppreference.com/w/cpp/types/numeric_limits/signaling_NaN

Reply:
A. to why? is to make sure the double is never used.

Reply:
I did read that page, added the includes, still no joy, had to run, Nils. Is the pragma what I’m missing?

Reply:
Considering there are a seemingly endless number of compiler errors I'd need more info :)

Reply:
OK, I figured it out. The page nils and I were referencing, besides showing what to include, shows the correct formulation is std::numeric_limits<double>::signaling_NaN(), not std::numeric_limits::signaling_nan().

Reply:
All you should need to do is
`#include <limits>`
and modify the statement to be 
`constexpr double d = std::numeric_limits<double>::signaling_NaN();`